### PR TITLE
Support for pre- and post-traversal

### PR DESCRIPTION
**Pull request by jwalton**
See #3.

### CODE REVIEW

**Comment by jwalton** [comment]
Nudge.  :)

**Comment by epoberezkin** [comment]
Sorry, I will merge over weekend. Could you please make it es5?

**Comment by jwalton** [comment]
No worries - if it were me it would take more nudges than that.  :)

Sorry; what did I do that's not es5?  Or are you meaning the example in the README.doc?

**Comment by epoberezkin** [comment]
E.g. {cb} should be {cb: cb} (it even feels weird now, doesn’t it :)

**Comment by epoberezkin** [comment]
Oh, that was readme... No, that’s fine.

**Comment by jwalton** [comment]
Haha.  I hear you.  I just started toying around with Typescript, and last week I ran into a weird TS bug that makes it so you can't inherit from `Error` when you're targeting ES3/ES5.  At first I was trying to come up with clever ways around it so I could support Node 4, but then I remembered the Node 4 was about to go EOL at the end of April.  Feels so strange that all these "new" javascript features aren't new anymore.  :P

File: index.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 var traverse = module.exports = function (schema, opts, cb) {
+  // Legacy support for v0.3.1 and earlier.
   if (typeof opts == 'function') {
     cb = opts;
     opts = {};
   }
-  _traverse(opts, cb, schema, '', schema);
+
+  cb = opts.cb || cb;
+  var pre = (typeof cb == 'function') ? cb : cb.pre || function() {};
```
Comment (epoberezkin) [inline index.js]: cb must be passed one way or another, so supplying an empty function here would hide an issue... I will remove it. Otherwise looks good.

Comment (jwalton) [reply index.js]: My thinking was you could call `traverse(schema, {cb: {post: () => ...}})` if you only want to do something in the post-traverse and don't care about the pre-traverse.

Comment (epoberezkin) [reply index.js]: I see... All right then.

**Pull request merged**